graph/src/data/store/mod.rs

```diff
@@ -31,51 +33,82 @@ pub enum Value {
-            (value, NonNullType(t)) => Value::from_query_value(value, t),
+            (value, NonNullType(t)) => Value::from_query_value(value, t)?,
 
             (query::Value::List(values), ListType(ty)) => Value::List(
                 values
                     .iter()
                     .map(|value| Self::from_query_value(value, ty))
-                    .collect(),
+                    .collect::<Result<Vec<_>, _>>()?,
             ),
 
             (query::Value::List(values), NamedType(n)) => Value::List(
                 values
                     .iter()
                     .map(|value| Self::from_query_value(value, &NamedType(n.to_string())))
-                    .collect(),
+                    .collect::<Result<Vec<_>, _>>()?,
             ),
-
+            (query::Value::Enum(e), NamedType(n)) => {
+                // Check if `ty` is a custom scalar type, otherwise assume it's
+                // just a string.
+                match n.as_str() {
+                    BYTES_SCALAR => Value::Bytes(scalar::Bytes::from_str(e)?),
+                    BIG_INT_SCALAR => Value::BigInt(scalar::BigInt::from_str(e)?),
+                    _ => Value::String(e.clone()),
+                }
+            }
             (query::Value::String(s), NamedType(n)) => {
                 // Check if `ty` is a custom scalar type, otherwise assume it's
                 // just a string.
                 match n.as_str() {
-                    BYTES_SCALAR => {
-                        Value::Bytes(scalar::Bytes::from_str(s).expect("Value is not a hex string"))
-                    }
-                    BIG_INT_SCALAR => {
-                        Value::BigInt(scalar::BigInt::from_str(s).expect("Value is not a number"))
-                    }
+                    BYTES_SCALAR => Value::Bytes(scalar::Bytes::from_str(s)?),
+                    BIG_INT_SCALAR => Value::BigInt(scalar::BigInt::from_str(s)?),
                     _ => Value::String(s.clone()),
                 }
             }
             (query::Value::Int(i), _) => Value::Int(
                 i.to_owned()
                     .as_i64()
-                    .expect("Unable to parse graphql_parser::query::Number into i64")
+                    .ok_or(QueryExecutionError::NamedTypeError("Int".to_string()))?
                     as i32,
             ),
             (query::Value::Float(f), _) => Value::Float(f.to_owned() as f32),
             (query::Value::Boolean(b), _) => Value::Bool(b.to_owned()),
             (query::Value::Null, _) => Value::Null,
-            _ => unimplemented!(),
-        }
+            _ => {
+                return Err(QueryExecutionError::AttributeTypeError(
+                    value.to_string(),
+                    ty.to_string(),
+                ))
+            }
+        })
+    }
+}
+
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let printable = match self {
```

Comment:
Wherever possible, it's good to not establish new names/variables. This one is a pretty simple case, so I'm ok with it. You could avoid it with
```
write!(f, "{}", match self {
  ...
})
```